src/AppContainer.js

```diff
@@ -27,6 +27,12 @@ const AppContainer = () => {
     }
   }, [escapePress]);
 
+  useEffect(() => {
```

Comment:
No need callback what. Within the overlay code, when ever setOverlayContent u can set the visible to true le. 

Also, can you move all the overlay related code from the app container to within the overlay component? It's logic that the parent dont need to know:

```
  useEffect(() => {
    if (escapePress) {
      setOverlayVisible(false);
    }
  }, [escapePress]);

  useEffect(() => {
    if (overlayContent !== null) {
      setOverlayVisible(true);
    }
  }, [overlayContent]); // callback when setOverlayContent is called

  const onOverlayTransitionEnded = () => {
    setOverlayContent(null);
  };
```

Reply:
shifted the related code. think might still need the callback as `in={isOverlayVisible}` is determining if the component is rendered at all. either we:

- `setOverlayVisible(true)` explicitly when activating the overlay onClicks in respective components. (initially solution, `setOverlayContent(<>content</>)` then `setOverlayVisible(true)`)
- maintain this callback based on when overlayContent is being set.
- change `in={isOverlayVisible}` to ` in={overlayContent !== null}`, this will result in no css transition, since rendering of component depends on if content is set, which also equates to =
- do away with CSSTransition entirely, so the component just toggles upfront with no transition. (this can adhere to suggested changes of doing away of callback on overlayContent set)

(Overlay.tsx)
![Screenshot 2020-05-11 at 1 42 11 PM](https://user-images.githubusercontent.com/4774314/81528109-85580a00-938e-11ea-86f4-07a3beed6c71.png)

OverlayContent (index.tsx)
![Screenshot 2020-05-11 at 1 42 19 PM](https://user-images.githubusercontent.com/4774314/81528102-82f5b000-938e-11ea-9f76-6d179464e82f.png)